src/storage/src/hummock/store/version.rs

```diff
@@ -610,20 +611,26 @@ impl HummockVersionReader {
                 );
 
                 let mut sstables = vec![];
-                for sstable_info in &level.table_infos[start_table_idx..=end_table_idx] {
-                    if sstable_info
-                        .table_ids
-                        .binary_search(&read_options.table_id.table_id)
-                        .is_err()
-                    {
-                        continue;
-                    }
-                    let sstable = self
-                        .sstable_store
-                        .sstable(sstable_info, &mut local_stats)
+                let fetch_meta_req = level.table_infos[start_table_idx..=end_table_idx]
+                    .iter()
+                    .filter(|sstable_info| {
+                        sstable_info
+                            .table_ids
+                            .binary_search(&read_options.table_id.table_id)
+                            .is_ok()
+                    })
+                    .collect_vec();
+                let fetch_meta_resp = try_join_all(fetch_meta_req.iter().map(|sstable_info| {
```

Comment:
With this PR, we only have meta parallel read per level. Can we optimize it further to allow parallel meta read for all SSTs in all levels?
1. Prune SSTs using `SstableInfo` for all levels in the 1st for loop to calculate `fetch_meta_req`
2. `try_join_all` the result of 1)

Reply:
> With this PR, we only have meta parallel read per level. Can we optimize it further to allow parallel meta read for all SSTs in all levels?
> 
> 1. Prune SSTs using `SstableInfo` for all levels in the 1st for loop to calculate `fetch_meta_req`
> 2. `try_join_all` the result of 1)

fixed